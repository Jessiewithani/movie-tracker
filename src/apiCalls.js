const baseUrl = 'https://api.themoviedb.org/3/movie/now_playing?api_key=';
const apiKey = '28964418fdafb10fc97bbbad131d01c3&language=en-US&page=1';

export const fetchMovies = async () => {
  const response = await fetch(`${baseUrl}${apiKey}`);
  const data = await response.json();
  return data
}