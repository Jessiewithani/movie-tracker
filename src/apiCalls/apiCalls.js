const baseUrl = 'https://api.themoviedb.org/3/movie/now_playing?api_key=';
const apiKey = '28964418fdafb10fc97bbbad131d01c3&language=en-US&page=1';

export const fetchMovies = async () => {
  const response = await fetch(`${baseUrl}${apiKey}`);
  const data = await response.json();
  const cleanData = data.results.map(movie => {
    const { id, title, poster_path, release_date, vote_average, overview } = movie
    movie = {
      movie_id: id,
      title: title,
      poster_path: poster_path,
      release_date: release_date,
      vote_average: vote_average,
      overview: overview
    }
    return movie
  })
  return cleanData;
}

export const postNewUser = async (newUser) => {
  const options = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(newUser)
  };

  let response = await fetch('http://localhost:3001/api/v1/users', options);

  return response;
}


export const loginUser = async (userInfo) => {
  const options = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(userInfo)
  };

  let response = await fetch('http://localhost:3001/api/v1/login', options);

  return response;
}

export const getFavorites = async (userId) => {
  let url = `http://localhost:3001/api/v1/users/${userId}/moviefavorites`;
  let response = await fetch(url);
  return response.json();
}

export const postFavorite = async (userId, movie) => {
  let url = `http://localhost:3001/api/v1/users/${userId}/moviefavorites`;
  let options = {
    method: 'POST',
    body: JSON.stringify(movie),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  let response = await fetch(url, options);
  console.log(response)

  if (!response.ok) {
    console.error('Could not add movie to favorites.')
  }
}

export const deleteFavorite = async (userId, movieId) => {
  let url = `http://localhost:3001/api/v1/users/${userId}/moviefavorites/${movieId}`
  let options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }

  let response = await fetch(url, options);
  console.log(response)
  if (!response.ok) {
      console.error('Could not delete movie from favorites.')
  }
}