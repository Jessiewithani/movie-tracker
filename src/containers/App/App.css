.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.App-link {
  color: #09d3ac;
}
