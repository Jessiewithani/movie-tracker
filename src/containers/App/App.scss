@import url('https://fonts.googleapis.com/css?family=Nanum+Gothic&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  height: 100vh;
}

.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  display: flex;
  align-items: center;
}

.App-link {
  color: #09d3ac;
}

.error-container {
  background-color: black;
  height: 20px;
}

.error {
  color: red;
  font-family: 'Josefin Slab', sans-serif;
  font-size: 1.2em;
}