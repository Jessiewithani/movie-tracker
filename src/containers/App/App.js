import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import './App.css';
import MoviesContainer from '../MoviesContainer/MoviesContainer'
import { saveMovies } from '../../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchMovies } from '../../apiCalls.js';

class App extends Component {


  async componentDidMount() {
    const { movies } = this.props; 
    console.log(movies)
    try {
      const movieData = await fetchMovies();
      await this.props.saveMovies(movieData)
    }

    catch {
    }
  }

  render() {
    return (
      <div className="App">
        <Route exact path='/' render={() => <MoviesContainer /> } />
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    saveMovies,
  }, dispatch)
)

const mapStateToProps = ({ movies }) => ({
  movies,
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
