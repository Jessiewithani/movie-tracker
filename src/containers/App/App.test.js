import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps }from './App';
import { fetchMovies } from '../../apiCalls/apiCalls';

jest.mock('../../apiCalls/apiCalls')

describe('App', () => {
  beforeEach(() => {
    fetchMovies.mockImplementation(() => {
      return Promise.resolve([{
        title: "Joker",
        id: 4,
        release_date: "2019-10-18"
      },
      {
        title: "Maleficent: Mistress of Evil",
        id: 56,
        release_date: "2019-10-04"
      },
      {
        title: "Zombieland: Double Tap",
        id: 70,
        release_date: "2019-10-18"
      }]);
    });
  })
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should retrieve movies after mounting', () => {
    shallow(<App/>);
    expect(fetchMovies).toHaveBeenCalled();
  })
    
})

describe('mapStateToProps In App', () => {
  it('should return an object with the user\'s info', () => {
    //SETUP
    const mockState = {
      currentUser: {name: 'Victor', email: 'Victor@gmail.com', password: 'Victor123'},
      setCurrentUser: 'SET_CURRENT_USER'
    }
    const expected = {
      currentUser: {name: 'Victor', email: 'Victor@gmail.com', password: 'Victor123'}
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  });
  it('should return an array with all the favorited movies', () => {
    const mockState = {
      favorites: [{
        title: "Maleficent: Mistress of Evil",
        id: 56,
        release_date: "2019-10-04"
      },
      {
        title: "Zombieland: Double Tap",
        id: 70,
        release_date: "2019-10-18"
      }],
      saveFavorites: 'SAVE_FAVORITES'
    }
    const expected = {
      favorites: [{
        title: "Maleficent: Mistress of Evil",
        id: 56,
        release_date: "2019-10-04"
      },
      {
        title: "Zombieland: Double Tap",
        id: 70,
        release_date: "2019-10-18"
      }]
    }
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  });
  it('should show the login status of the user', () => {
    const mockState = {
      isLoggedIn: true,
      login: 'LOGIN'

    }
    const expected = {
      isLoggedIn: true
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  });
  it('should return an array of movies', () => {
    const mockState = {
      movies: [{
        title: "Joker",
        id: 4,
        release_date: "2019-10-18"
      },
      {
        title: "Maleficent: Mistress of Evil",
        id: 56,
        release_date: "2019-10-04"
      },
      {
        title: "Zombieland: Double Tap",
        id: 70,
        release_date: "2019-10-18"
      }],
      saveMovies: 'SAVE_MOVIES'
    }

    const expected = {
      movies: [{
        title: "Joker",
        id: 4,
        release_date: "2019-10-18"
      },
      {
        title: "Maleficent: Mistress of Evil",
        id: 56,
        release_date: "2019-10-04"
      },
      {
        title: "Zombieland: Double Tap",
        id: 70,
        release_date: "2019-10-18"
      }]
    }
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  })
})

describe('mapDispathToProps in App', () => {
  
})