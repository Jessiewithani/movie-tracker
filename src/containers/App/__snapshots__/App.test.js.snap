// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Connect(NavBar) />
  <div
    className="error-container"
  >
    <h2
      className="error"
    />
  </div>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(MovieInfo)",
        "type": [Function],
      }
    }
    exact={true}
    path="/movies/:id"
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/new-user"
    render={[Function]}
  />
</div>
`;
